Snowflake.Data/Client/SnowflakeDbConnection.cs

```diff
@@ -86,17 +99,53 @@ public override string DataSource
             }
         }
 
-        public override string ServerVersion => _connectionState == ConnectionState.Open ? SfSession.serverVersion : "";
+        public override string ServerVersion => IsOpen() ? SfSession.serverVersion : "";
 
         public override ConnectionState State => _connectionState;
 
+        private object? GetTransactionId()
+        {
+            if (!IsOpen())
+            {
+                return DBNull.Value;
+            }
+            using (IDbCommand command = CreateCommand())
+            {
+                command.CommandText = "SELECT CURRENT_TRANSACTION()";

```

Comment:
We do a roundtrip to SF here every time this function is invoked. I would say, that I would rather expect the `GetTransactionId` call did not go to server. But I'm not fully convinced, maybe this is a premature optimization...

Reply:
The problem is not only user can run SnowflakeDbConnection.BeginDbTransaction but can also create a DbCommand, provide it's CommandText = "BEGIN" and trigger the command without this operation outcome being noticed by SnowflakeDbConnection.
I had same worries here but stateful connection might be a thing to discuss in the future.
We need also to remember that right now there is just one transaction at a time available but introducing a new feature in the Snowflake DB can lead to related changes in the SnowflakeDbConnection if we make it stateful. Not sure about this idea then.

Reply:
I see your concerns, but on the other hand - it's private implementation, even if something changes we can refactor it. I thought about caching this value after first call to getting transaction id and clearing on transaction end. See, my main concern is that it's easy for instance to have a client write code like this:

```
for (record in records) {
    log.debug("session: {}, transactionId: {}, record:", getSessionId(), getTransactionId(), record)
}
```

And now, even if debug is not logged, we still have a lot of roundtrips.

Reply:
Transaction can be started without SnowflakeDbConnection being aware of it. Won't do it this way